packages/toolkit/src/view/model/ModelHubListPageMainView.tsx

```diff
@@ -17,52 +15,125 @@ export const ModelHubListPageMainView = (
-      ? modelsWatchState.isLoading
-      : false;
+    !models.isFetched ||
+    models.isLoading ||
+    models.isFetching ||
+    models.isFetchingNextPage;
 
   /* -------------------------------------------------------------------------
    * Render
    * -----------------------------------------------------------------------*/
 
   return (
     <div className="flex flex-col">
-      <div className="mb-8">
+      <div className="mb-4 flex flex-row items-end gap-x-3">
+        <div className="flex flex-col gap-y-2.5">
+          <p className="text-semantic-fg-primary product-body-text-3-semibold">
+            Search Models
+          </p>
+          <div className="mt-auto flex flex-row gap-x-4">
+            <Input.Root className="flex-1">
+              <Input.LeftIcon>
+                <Icons.SearchSm className="my-auto h-4 w-4 stroke-semantic-fg-primary" />
+              </Input.LeftIcon>
+              <Input.Core
+                value={searchInputValue ?? ""}
+                placeholder="Search..."
+                onChange={(event) => {
+                  setSearchInputValue(event.target.value);
+                  debouncedSetSearchCode(event.target.value);
+                }}
+              />
+            </Input.Root>
+          </div>
+        </div>
+        <div className="flex min-w-52 flex-col gap-y-2.5">
+          <p className="text-semantic-fg-primary product-body-text-3-semibold">
+            Visibility
+          </p>
+          <Select.Root
+            value={selectedVisibilityOption}
+            onValueChange={(value) => {
+              setSelectedVisibilityOption(value as Visibility);
+            }}
+          >
+            <Select.Trigger className="mt-auto w-full">
+              <Select.Value />
+            </Select.Trigger>
+            <Select.Content>
+              <Select.Group>
+                <Select.Item value="VISIBILITY_UNSPECIFIED" label="All" />
+                <Select.Item value="VISIBILITY_PUBLIC" label="Public" />
+                <Select.Item value="VISIBILITY_PRIVATE" label="Private" />
+              </Select.Group>
+            </Select.Content>
+          </Select.Root>
+        </div>
         <Button
-          className="gap-x-2"
+          className="ml-auto"
           variant="primary"
           size="lg"
           onClick={() => {
             router.push(`/${entity}/models/create`);
           }}
         >
-          <Icons.Plus className="h-4 w-4 stroke-semantic-bg-primary" />
-          Add Model
+          Create a Model
         </Button>
       </div>
-      <ModelsTable
-        models={models.isSuccess ? models.data : []}
-        modelsWatchState={
-          modelsWatchState.isSuccess ? modelsWatchState.data : {}
+      <ModelsList
+        models={
+          models.isSuccess
+            ? models.data?.pages[pageNumber]
+              ? models.data?.pages[pageNumber].models
+              : []
+            : []
```

Comment:
@orangecoloured react-query will make sure once the models is success, the data is there, so you don't need to use `Optional chaining` here

Reply:
Do you mean to have it like this?
```
models={
  models.isSuccess
    ? models.data.pages[pageNumber]?.models
    : []
}
```

Reply:
yes exactly